src/main/java/bg/codexio/springframework/data/jpa/requery/adapter/GraphQLHttpFilterAdapter.java

```diff
@@ -0,0 +1,281 @@
+        };
+
+        for (var suffix : suffixes) {
+            if (name.endsWith(suffix)) {
+                return name.substring(0, name.length() - suffix.length());
+            }
+        }
+
+        return name;
+    }
+
+    private Object extractValue(Value<?> value) {
+        if (value instanceof StringValue) {
+            return ((StringValue) value).getValue();
+        } else if (value instanceof IntValue) {
+            return ((IntValue) value).getValue();
+        } else if (value instanceof BooleanValue) {
+            return ((BooleanValue) value).isValue();
+        } else if (value instanceof FloatValue) {
+            return ((FloatValue) value).getValue();
+        } else if (value instanceof EnumValue) {
+            return ((EnumValue) value).getName();
+        } else if (value instanceof ArrayValue) {
+            return ((ArrayValue) value).getValues().stream().map(this::extractValue).toList();
+        } else if (value instanceof ObjectValue) {
+            return handleComplexObject((ObjectValue) value);
+        } else {
+            return null;
+        }
+    }
+
+    private Map<String, Object> handleComplexObject(ObjectValue value) {
+        var result = new HashMap<String, Object>();
+        for (var field : value.getObjectFields()) {
+            result.put(field.getName(), field.getValue());
+        }
+        return result;
+    }
+
+    private FilterOperation getOperationFromArgument(String argumentName) {
+        if (argumentName.endsWith("_gt")) {
+            return FilterOperation.GT;
+        } else if (argumentName.endsWith("_gte")) {
+            return FilterOperation.GTE;
+        } else if (argumentName.endsWith("_lt")) {
+            return FilterOperation.LT;
+        } else if (argumentName.endsWith("_lte")) {
+            return FilterOperation.LTE;
+        } else if (argumentName.endsWith("_in")) {
+            return FilterOperation.IN;
+        } else if (argumentName.endsWith("_not_in")) {
+            return FilterOperation.NOT_IN;
+        } else if (argumentName.endsWith("_contains")) {
+            return FilterOperation.CONTAINS;
+        } else if (argumentName.endsWith("_starts_with")) {
+            return FilterOperation.BEGINS_WITH;
+        } else if (argumentName.endsWith("_ends_with")) {
+            return FilterOperation.ENDS_WITH;
+        } else if (argumentName.endsWith("_empty")) {
+            return FilterOperation.EMPTY;
+        } else if (argumentName.endsWith("_not_empty")) {
+            return FilterOperation.NOT_EMPTY;
+        } else if (argumentName.endsWith("_begins_with_caseins")) {
+            return FilterOperation.BEGINS_WITH_CASEINS;
+        } else if (argumentName.endsWith("_ends_with_caseins")) {
+            return FilterOperation.ENDS_WITH_CASEINS;
+        } else if (argumentName.endsWith("_contains_caseins")) {
+            return FilterOperation.CONTAINS_CASEINS;
+        }
+        return FilterOperation.EQ;
+    }
```

Comment:
As this and `extractName` methods are both using these values might be good to take this out into a static map `OPERATION_MAP` and use it instead. I can imagine a map where the key is that string and the value is the FilterOperation and have something like this here
```suggestion
    private FilterOperation getOperationFromArgument(String argumentName) {
        return OPERATION_MAP.entrySet()
                            .stream()
                            .filter(entry -> argumentName.endsWith(entry.getKey()))
                            .findFirst()
                            .map(Map.Entry::getValue)
                            .orElse(FilterOperation.EQ);
    }
```

Reply:
done